packages/thumbprint-react/components/Popover/index.jsx

```diff
@@ -0,0 +1,219 @@
+            >
+                {children}
+
+                <div className={styles.closeButton}>
+                    <TextButton
+                        accessibilityLabel="Close popover"
+                        iconLeft={<NavigationCloseTiny className={styles.closeButtonIcon} />}
+                        theme="inherit"
+                        onClick={onCloseClick}
+                    />
+                </div>
+
+                <div
+                    className={classNames({
+                        [styles.nubbin]: true,
+                        [styles.nubbinTop]: startsWith(placement, 'bottom'),
+                        [styles.nubbinBottom]: startsWith(placement, 'top'),
+                        [styles.nubbinLeft]: startsWith(placement, 'right'),
+                        [styles.nubbinRight]: startsWith(placement, 'left'),
+                    })}
+                    ref={arrowProps.ref}
+                    style={arrowProps.style}
+                />
+            </div>
+        )}
+    </Popper>
+);
+
+const Popover = ({ children, launcher, position, isOpen, onCloseClick, container }) => {
+    // Appends the tooltip right before `</body>` when true.
+    const shouldDisplace = container === 'body';
+
+    const handleKeyUp = event => {
+        if (event.keyCode === ESC_KEY) {
+            onCloseClick();
+        }
+    };
+
+    useEffect(() => {
+        document.addEventListener('keyup', handleKeyUp);
+        return () => {
+            document.removeEventListener('keyup', handleKeyUp);
+        };
+    });
+
+    Popover.handleClickOutside = () => onCloseClick();
+
+    const popoverContent = (
+        <PopoverContent position={position} isOpen={isOpen} onCloseClick={onCloseClick}>
+            {children}
+        </PopoverContent>
+    );
+
+    return (
+        <Manager>
+            <Reference>{({ ref }) => launcher({ ref })}</Reference>
+            {canUseDOM &&
+                (shouldDisplace
+                    ? ReactDOM.createPortal(popoverContent, document.body)
+                    : popoverContent)}
+        </Manager>
+    );
+};
+
+Popover.propTypes = {
+    /**
+     * Contents for the Popover. Usually a `PopoverTitle`, `PopoverBody`, and `PopoverPrimaryButton`
+     */
+    children: PropTypes.node,
+    /**
+     * A function that renders JSX and receives an object with `ref`.
+     * All of these props must be added to the component within the render prop.
+     */
+    launcher: PropTypes.func.isRequired,
+    /**
+     * Position of popover relative to the launcher.
+     */
+    position: PropTypes.oneOf([
+        'top-start',
+        'top',
+        'top-end',
+        'bottom-start',
+        'bottom',
+        'bottom-end',
+        'left-start',
+        'left',
+        'left-end',
+        'right-start',
+        'right',
+        'right-end',
+    ]),
+    /**
+     * Whether or not the popover is visible.
+     */
+    isOpen: PropTypes.bool.isRequired,
```

Comment:
Most (all?) of our controlled components make this not required and then default to `false` (or `undefined` for text inputs).